manifest/github.go

```diff
@@ -193,27 +195,61 @@ func (ghp GitHubProvider) manifestsFromWildcard() ([]*Manifest, error) {
 		opts.Page = resp.NextPage
 	}
 
+	return ghp.appendRepositoriesToManifestInParallel(repos)
+}
+
+func (ghp GitHubProvider) appendRepositoriesToManifestInParallel(repos []*github.Repository) ([]*Manifest, error) {
 	var manifests []*Manifest
+	var errs []error
+	var wg sync.WaitGroup
+	var mu sync.RWMutex
+	wg.Add(len(repos))
 
 	for _, repo := range repos {
-		manifest, err := gitProviderFromGitHub(ghp.ctx, *repo.CloneURL, ghp.mopts...)
-		if err != nil {
-			return nil, err
-		}
+		go func(repo *github.Repository) {
+			defer wg.Done()
+			manifest, err := gitProviderFromGitHub(ghp.ctx, *repo.CloneURL, ghp.mopts...)
+			if err != nil {
+				mu.Lock()
+				errs = append(errs, err)
+				mu.Unlock()
+				return
+			}
 
-		// Populate with GitHub API-centric information
-		if repo.Description != nil {
-			manifest.Description = *repo.Description
-		}
+			// Populate with GitHub API-centric information
+			if repo.Description != nil {
+				manifest.Description = *repo.Description
+			}
 
-		t, _, _, err := unikraft.GuessTypeNameVersion(*repo.Name)
-		if err != nil {
-			return nil, err
-		}
+			t, _, _, err := unikraft.GuessTypeNameVersion(*repo.Name)
+			if err != nil {
+				mu.Lock()
+				errs = append(errs, err)
+				mu.Unlock()
+				return
+			}
 
-		manifest.Type = t
+			manifest.Type = t
 
-		manifests = append(manifests, manifest)
+			mu.Lock()
+			manifests = append(manifests, manifest)
+			mu.Unlock()
+		}(repo)
+	}
+
+	wg.Wait()
+
+	// combine errors into a single error
+	if len(errs) > 0 {
+		var msg string
+		for _, err := range errs {
+			if msg == "" {
+				msg = err.Error()
+			} else {
+				msg = fmt.Sprintf("%s\n%s", msg, err.Error())
+			}
+		}
+		return nil, fmt.Errorf("%w", errors.New(msg))
 	}
```

Comment:
Since the project uses Go 1.20, we can use the [`errors.Join`](https://pkg.go.dev/errors#Join) method to wrap errors.